### PR TITLE
[SLT-157] feat(rest-api): Refactor for RESTful architectural standardization

### PR DESCRIPTION
**Pull request by abtestingalpha**
**Description**
Refactors app to support MVC architecture and RESTful responses.

* architectural standardization with routes and controllers
* local development enhancements: `yarn dev` hot reload
* error handling middleware
* validates inputs, no silent response errors

## Summary by CodeRabbit

- **New Features**
	- Introduced new controllers for managing bridge transactions, swap transactions, and retrieving transaction statuses.
	- Added a new endpoint for accessing a list of tokens.
	- Implemented a new service for interacting with the Synapse cross-chain protocol.
	- Added a Jest configuration for improved testing capabilities.

- **Improvements**
	- Enhanced input validation for various API endpoints.
	- Refactored the application structure for better maintainability and readability.
	- Added a development script to streamline the development workflow.
	- Centralized valid bridge module definitions for consistency.
	- Updated dependencies for improved functionality and security.

- **Bug Fixes**
	- Addressed issues related to validation error handling and response clarity.

- **Documentation**
	- Updated constants and utility functions for better clarity and usage.

### CODE REVIEW

File: packages/rest-api/src/routes/tokenListRoute.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import express from 'express'
+
+import { tokenListController } from '../controllers/tokensListController'
+
+const router = express.Router()
+
+router.get('/', tokenListController)
```
[the remark this answers is not in the document]
Comment (abtestingalpha) [reply packages/rest-api/src/routes/tokenListRoute.ts:7]: validation is in each of the routes

File: packages/rest-api/tsconfig.json [hunk 2]
```diff
@@ -20,7 +13,7 @@
     // linter checks for common issues
     "noImplicitReturns": true,
     "noFallthroughCasesInSwitch": true,
-    "noUnusedLocals": true,
+    "noUnusedLocals": false,
```
[the remark this answers is not in the document]
Comment (abtestingalpha) [reply packages/rest-api/tsconfig.json:16]: unused variables are caught during linting. enabling this to be true during development process breaks builds when experimenting.

File: packages/rest-api/src/app.ts [hunk 3]
```diff
@@ -1,672 +1,13 @@
...
-import { SynapseSDK } from '@synapsecns/sdk-router'
-import { BigNumber } from '@ethersproject/bignumber'
-import { formatUnits, parseUnits } from '@ethersproject/units'
 import express from 'express'
 
-import * as tokensList from './constants/bridgeable'
-import { CHAINS_ARRAY } from './constants/chains'
-import { Chain } from './types'
-import { BRIDGE_MAP } from './constants/bridgeMap'
+import routes from './routes'
```
[the remark this answers is not in the document]
Comment (abtestingalpha) [reply packages/rest-api/src/app.ts:3]: it's there in `./routes/index.ts` and importing from `./routes` is shorthand for that file

File: packages/rest-api/src/controllers/getBridgeTxStatusController.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+          }
+        `
+
+      const graphqlResponse = await fetch(graphqlEndpoint, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ query: graphqlQuery }),
+      })
```
[the remark this answers is not in the document]
Comment (abtestingalpha) [reply packages/rest-api/src/controllers/getBridgeTxStatusController.ts:50]: Worth doing if he have more graphql queries; right now theres just this one.

File: packages/rest-api/src/controllers/bridgeController.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+    const amountInWei = parseUnits(amount.toString(), fromTokenInfo.decimals)
+
+    const resp = await Synapse.allBridgeQuotes(
+      Number(fromChain),
+      Number(toChain),
+      fromTokenInfo.address,
+      toTokenInfo.address,
+      amountInWei
+    )
```
[the remark this answers is not in the document]
Comment (abtestingalpha) [reply packages/rest-api/src/controllers/bridgeController.ts:25]: They are both wrapped in try catch

**Comment by Defi-Moses** [comment]
All looks good to me, code is significantly cleaner. Some smaller questions I have for the next iteration:
- should we have a basic set of tests to protect the deployment branch (master)
- Is there a simple way we can try to load test?

**Review by Defi-Moses — APPROVED** [review]

**Pull request merged**